Remotion/ObjectBinding/Web/UI/Controls/BocListImplementation/Rendering/BocListValidationSummaryRenderer.cs

```diff
@@ -0,0 +1,113 @@
+    {
+      if (arguments.ScreenReaderOnly)
+      {
+        renderingContext.Writer.AddAttribute(HtmlTextWriterAttribute.Class, _bocListCssClassDefinition.CssClassScreenReaderText);
+      }
+
+      renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Ul);
+
+      foreach (var validationFailureWithLocation in arguments.ValidationFailures)
+        RenderValidationFailure(renderingContext, validationFailureWithLocation, arguments.RowIndex, arguments.ColumnIndexProvider);
+
+      renderingContext.Writer.RenderEndTag();
+    }
+
+    private void RenderValidationFailure (
+        BocRenderingContext<IBocList> renderingContext,
+        BocListValidationFailureWithLocationInformation validationFailureWithLocation,
+        int rowIndex,
+        IBocListColumnIndexProvider columnIndexProvider)
+    {
+      if (_renderingFeatures.EnableDiagnosticMetadata)
+      {
+        var rowObjectWithIdentity = validationFailureWithLocation.RowObject as IBusinessObjectWithIdentity;
+
+        renderingContext.Writer.AddAttribute(DiagnosticMetadataAttributesForObjectBinding.BocListValidationFailureMessage, validationFailureWithLocation.Failure.ErrorMessage);
+        renderingContext.Writer.AddAttribute(DiagnosticMetadataAttributesForObjectBinding.BocListValidationFailureSourceRow, rowObjectWithIdentity?.UniqueIdentifier ?? string.Empty);
+        renderingContext.Writer.AddAttribute(DiagnosticMetadataAttributesForObjectBinding.BocListValidationFailureSourceColumn, validationFailureWithLocation.ColumnDefinition?.ItemID ?? string.Empty);
+      }
+
+      renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Li);
+
+      var columnDefinition = validationFailureWithLocation.ColumnDefinition;
+      if (columnDefinition != null)
+      {
+        var visibleColumnIndex = columnIndexProvider.GetVisibleColumnIndex(columnDefinition);
+        Assertion.IsTrue(visibleColumnIndex != -1, "visibleColumnIndex != -1");
+
+        var validationMarkerCellID = BocRowRenderer.GetCellIDForValidationMarker(renderingContext.Control, rowIndex, visibleColumnIndex);
+        renderingContext.Writer.AddAttribute(HtmlTextWriterAttribute.Href, $"#{validationMarkerCellID}");
+        renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.A);
+
+        var columnTitle = columnDefinition.ColumnTitle.IsEmpty
+            ? WebString.CreateFromText($"Column {visibleColumnIndex}")
+            : columnDefinition.ColumnTitle;
+        columnTitle.WriteTo(renderingContext.Writer);
+        renderingContext.Writer.WriteEncodedText(": ");
+      }
+      else
+      {
+        renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Span);
+
+        var validatedProperty = validationFailureWithLocation.Failure.ValidatedProperty;
+        if (validatedProperty != null)
+        {
+          renderingContext.Writer.WriteEncodedText(validatedProperty.DisplayName);
+          renderingContext.Writer.WriteEncodedText(": ");
+        }
```

Comment:
I realized something: we may want to to move DisplayName into WebString way earlier. since it's probably already commited, this could be a separate cleanup task. It's just something we cleaned up a lot and finding this now feels weird.

Reply:
I don't think I understand. `validatedProperty` is a `IBusinessObjectProperty` so the `DisplayName` property is from there. We didn't even touch this class here so I don't know what you mean by "way earlier"